File: framework/languages/dotnet/e2e/SpotService/Client/Scenarios/SmD13Scenario.cs
using SpotService.Client;
using SpotService.Shared;
using Systems.Zlink.Stream.Connector.Contracts;

namespace SpotService.Client.Scenarios;

internal static class SmD13Scenario
{
    public static async Task RunAsync(string sessionAStreamEndpoint)
    {
        await using var stream = await SpotActorRequestSupport.ConnectAndAuthWithRetryAsync(
            sessionAStreamEndpoint,
            new AuthReq("actor-sm-d13", "heartbeat", "play-a"),
            TimeSpan.FromSeconds(5),
            heartbeat: new ZlinkStreamHeartbeatOptions
            {
                Enabled = true,
                Interval = TimeSpan.FromMilliseconds(200),
                Timeout = TimeSpan.FromSeconds(2),
            });
        await Task.Delay(600);
        ScenarioAssert.That(stream.IsConnected, "SM-D13 heartbeat-enabled stream disconnected.");
        Console.WriteLine("operation SpotService.sm-d13 passed");
    }
}

File: framework/languages/java/e2e/SpotService/Client/src/main/java/systems/zlink/e2e/spotservice/client/Scenarios/SmD13Scenario.java
package systems.zlink.e2e.spotservice.client.Scenarios;

import systems.zlink.e2e.spotservice.client.Support.GatewayScenarioClient;

public final class SmD13Scenario {
    private SmD13Scenario() {
    }

    public static void run(GatewayScenarioClient gateway) {
        gateway.runMode("stream-heartbeat");
    }
}
